websocket/src/test/java/org/nanohttpd/junit/protocols/websockets/WebSocketFrameTest.java

```diff
@@ -0,0 +1,140 @@
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+import java.io.ByteArrayInputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.charset.CharacterCodingException;
+import java.util.ArrayList;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.nanohttpd.protocols.websockets.OpCode;
+import org.nanohttpd.protocols.websockets.WebSocketException;
+import org.nanohttpd.protocols.websockets.WebSocketFrame;
+
+public class WebSocketFrameTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testReadEOFException() throws IOException {
+
+        // Arrange
+        final byte[] byteArray = {0};
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+
+        // Act
+        thrown.expect(EOFException.class);
+        WebSocketFrame.read(in);
+
+        // Method is not expected to return due to exception thrown
+    }
+
+    @Test
+    public void testReadWebSocketExceptionMaxFrameLength() throws IOException {
+
+        // Arrange
+        final byte[] byteArray = {-128, -1, 0, 0, 0, 1, 47, 120, -16, 16};
+
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        WebSocketFrame.read(in);
+    }
+
+    @Test
+    public void testReadWebSocketException4ByteLength() throws Exception {
+
+        // Arrange
+        final byte[] byteArray = {-128, -1, -128, 0, 0, 0, 47, 120, -16, 16
+        };
+
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        WebSocketFrame.read(in);
+    }
+
+    @Test
+    public void testReadWebSocketException2ByteLength() throws Exception {
+
+        // Arrange
+        final byte[] byteArray = {-59, -128, 126, 0, 0, 32, -1, -128, 4, -128};
+
+        // Arrange
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray, 1, 9);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        WebSocketFrame.read(in);
+
+    }
+
+    @Test
+    public void testReadPayloadInfoReservedBitsNotZero() throws Throwable {
+
+        // Arrange
+        final byte[] byteArray = {24, 127, 0, 0, 0, 0, 0, -128, 0, 1};
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+        final ArrayList<WebSocketFrame> fragments = new ArrayList<WebSocketFrame>();
+        final WebSocketFrame objectUnderTest = new WebSocketFrame(OpCode.Close, fragments);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        objectUnderTest.read(in);
```

Comment:
There are only exception tests? I'd expect a test for a positive test as well.

Reply:
The positive test is already covered!

Reply:
By a test from another file?

Reply:
yes